src/Lean/Server/InfoUtils.lean

```diff
@@ -150,13 +150,21 @@ partial def InfoTree.hoverableInfoAt? (t : InfoTree) (hoverPos : String.Pos) (in
     unless (info matches Info.ofFieldInfo _ || info.toElabInfo?.isSome) && info.contains hoverPos includeStop do
       return results
     let r := info.range?.get!
-    let priority :=
-      if r.stop == hoverPos then
-        0  -- prefer results directly *after* the hover position (only matters for `includeStop = true`; see #767)
-      else if info matches .ofTermInfo { expr := .fvar .., .. } then
-        0  -- prefer results for constants over variables (which overlap at declaration names)
-      else 1
+    let priority := (
+      -- prefer results directly *after* the hover position (only matters for `includeStop = true`; see #767)
+      if r.stop == hoverPos then 0 else 1,
+      -- relying on the info tree structure is _not_ sufficient for choosing the smallest surrounding node:
+      -- `⟨x⟩` expands to an application of a canonical syntax with the span of the anonymous constructor to `x`,
+      -- i.e. there are two info tree siblings whose spans are not disjoint and we should choose the smaller node
+      -- surrounding the cursor
+      Int.negOfNat (r.stop - r.start).byteIdx,
+      -- prefer results for constants over variables (which overlap at declaration names)
+      if info matches .ofTermInfo { expr := .fvar .., .. } then 0 else 1)
     [(priority, ctx, info)]) |>.getD []
+  -- sort results by lexicographical priority
+  let _ := @lexOrd
+  let _ := @leOfOrd.{0}
+  let _ := @maxOfLe
```

Comment:
This code is getting a bit creative after introduction of `Max`. Note the fun universe annotation: the let binding is not actually used in the final term, presumably because it was unfolded in `Decidable`.
Should these definitions be scoped instances instead?